### PR TITLE
Merge rmdatatable

### PR DESCRIPTION
**Pull request by JonasGeschke**
@KevCaz are you ready to merge the two branches so the master it up-to-date?

### CODE REVIEW

**Comment by ibartomeus** [comment]
@KevCaz I'll accept the pull request as soon as you give your blessing.

**Comment by KevCaz** [comment]
I'm not there yet, 'll add some testing first! :smile:

**Comment by JonasGeschke** [comment]
Ah alright. Sorry, didnt want to put pressure on you!

**Comment by KevCaz** [comment]
No worries! 

I removed `data.table`! I still have to re-code the all the tests, I would like to do so before we merge this.

**Comment by KevCaz** [comment]
I am finishing to create the print methods for all outputs and then I start the new set of tests.

**Comment by JonasGeschke** [comment]
alright. anything we can do to help you out?

File: NEWS.md [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+  - `spp_taxonconcept()` now includes an auto-pagination.
+  - Functions now returns S3 objects for which a print method is defined for all
+  these objects:
+    - `spp_raw` class is defined for all functions including the argument `raw`,
+    it returns the parsed output from the API as a list object.  
+    - `spp_taxonconcept` returns an object of class `spp_spp_taxon`
+    - `spp_cites_legislation` returns an object of class `spp_cites_leg`
+    - `spp_eu_legislation` returns an object of class `spp_eu_leg`
+    - `spp_distributions` returns an object of class `spp_distr`
+    - `spp_references` returns an object of class `spp_distr`
```
Comment (JonasGeschke) [inline NEWS.md]: should be `spp_refs`

File: NEWS.md [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
-  - `spp_taxonconcept()`
-  - outputs:
-    - spp_raw
+
+  - `spp_taxonconcept()` now includes an auto-pagination.
+  - Functions now returns S3 objects for which a print method is defined for all
+  these objects:
+    - `spp_raw` class is defined for all functions including the argument `raw`,
+    it returns the parsed output from the API as a list object.  
+    - `spp_taxonconcept` returns an object of class `spp_spp_taxon`
```
Comment (JonasGeschke) [inline NEWS.md]: double `spp_spp_` ?

**Comment by KevCaz** [comment]
So far so good! I should be done by the end of the day and then I'll need help to review what I've done and to edit the documentation!

File: NEWS.md (hunk 2 above)
Comment (KevCaz) [reply NEWS.md]: done! `spp_taxon`

File: NEWS.md (hunk 1 above)
Comment (KevCaz) [reply NEWS.md]: done! thanks.

**Comment by KevCaz** [comment]
We can merge this branch, review what have been done and edit the documentation. 
That said, I think we should rather wait for the second review because if extra changes are required, this means extra work. Anyway, I'll wait the second review to complete the unit testing.

**Pull request merged**